x-pack/metricbeat/module/azure/client.go

```diff
@@ -195,10 +195,15 @@ func (client *Client) InitResources(fn mapResourceMetrics) error {
 // |
 //
 // In this case, the API will return one metric value.
-func buildTimespan(referenceTime time.Time, timeGrain string, collectionPeriod time.Duration) string {
-	timespanDuration := max(asDuration(timeGrain), collectionPeriod)
+func buildTimespan(referenceTime time.Time, timeGrain string, config Config) string {
+	timespanDuration := max(asDuration(timeGrain), config.Period) // Period is the collection period.
 
-	endTime := referenceTime
+	// The end time is the reference time in most cases.
+	//
+	// However, if the Azure service publishes the metric values with
+	// a delay, we can translate the reference time to compensate
+	// for the latency.
+	endTime := referenceTime.Add(config.Latency * -1)
```

Comment:
I see that Add function will return the t+d time , so here seems that we always compute: 
[referenceTime - config.Latency] or because referenceTime is (from [here](https://github.com/elastic/beats/blob/812b877e011d702a6cad487d0ad17e5e9e1f09ce/x-pack/metricbeat/module/azure/azure.go#L165)) always time.Now, we have [now - config.Latency].

To the question: shall we check if config.Latency is configured and if not have a default value to 0?